src/invoke.rs

```diff
@@ -97,27 +100,42 @@ impl Cmd {
         };
 
         // re-assemble the args, to match the order given on the command line
-        let args: Vec<ScVal> = if self.args_xdr.is_empty() {
-            self.args
-                .iter()
-                .zip(inputs.iter())
-                .map(|(a, input)| strval::from_string(a, &input.type_))
-                .collect::<Result<Vec<_>, _>>()?
-        } else {
-            self.args_xdr
-                .iter()
-                .map(|a| match base64::decode(a) {
+        let indexed_args: Vec<(usize, Arg)> = matches
+            .indices_of("args")
+            .unwrap_or_default()
+            .zip(self.args.iter())
+            .map(|(a, b)| (a, Arg::Arg(b.to_string())))
+            .collect();
+        let indexed_args_xdr: Vec<(usize, Arg)> = matches
+            .indices_of("args-xdr")
+            .unwrap_or_default()
+            .zip(self.args_xdr.iter())
+            .map(|(a, b)| (a, Arg::ArgXDR(b.to_string())))
+            .collect();
+        let mut all_indexed_args: Vec<(usize, Arg)> = [indexed_args, indexed_args_xdr].concat();
+        all_indexed_args.sort_by(|a, b| a.0.cmp(&b.0));
+
+        if all_indexed_args.len() > inputs.len() {
+            return Err(Error::TooManyArguments);
+        }
+
+        let parsed_args: Vec<ScVal> = all_indexed_args
+            .iter()
+            .zip(inputs.iter())
+            .map(|(arg, input)| match &arg.1 {
+                Arg::ArgXDR(s) => match base64::decode(s) {
                     Err(_) => Err(StrValError::InvalidValue),
                     Ok(b) => ScVal::from_xdr(b).map_err(StrValError::Xdr),
```

Comment:
Unrelated Nit/Suggestion: We can actually use `ScVal::from_xdr_base64` now, and it will decode and parse in one step.